ui/rogue/combat/sim.ts

```diff
@@ -187,12 +188,41 @@ const SPEC_CONFIG = registerSpecConfig(Spec.SpecCombatRogue, {
 	],
 });
 
+function getMasterySoftCapConfig(): StatCap {
+	const masteryRatingBreakpoints = [];
+	const masteryPercentPerPoint = Mechanics.masteryPercentPerPoint.get(Spec.SpecCombatRogue)!;
+	for (let masteryPercent = 16; masteryPercent <= 200; masteryPercent++) {
+		masteryRatingBreakpoints.push((masteryPercent / masteryPercentPerPoint) * Mechanics.MASTERY_RATING_PER_MASTERY_POINT);
+	}
+
+	return StatCap.fromStat(Stat.StatMasteryRating, {
+		breakpoints: masteryRatingBreakpoints,
+		capType: StatCapType.TypeThreshold,
+		postCapEPs: [0],
+	});
+}
+
+function AddOrRemoveMasteryBreakpoint(optimizer: ReforgeOptimizer, isShown: boolean): void {
+	if (isShown == true) {
+		optimizer.softCapsConfig.push(getMasterySoftCapConfig());
+	}
+	else
+	{
+		optimizer.softCapsConfig.splice(2, 1);
+	}
+}
+
 export class CombatRogueSimUI extends IndividualSimUI<Spec.SpecCombatRogue> {
 	constructor(parentElem: HTMLElement, player: Player<Spec.SpecCombatRogue>) {
 		super(parentElem, player, SPEC_CONFIG);
 
 		player.sim.waitForInit().then(() => {
-			new ReforgeOptimizer(this);
+			const optimizer = new ReforgeOptimizer(this);
+			AddOrRemoveMasteryBreakpoint(optimizer, this.sim.getShowExperimental())
```

Comment:
Instead of binding a changeEmitter here I would just use the optimizer's `updateSoftCaps` callback function to enrich the Mastery breakpoints. You can check the `mage/fire/sim.ts` for an example. We modify it there, but you can also straight up just add an entry

Reply:
**If** you wanted multiple EPs, you could also set that up. Example in the `warrior/fury/sim.ts`

Reply:
Yeah I've gone back and fourth on having multiple EPs. After trying what I had on both T11 and T12 setups, I've settled on making one specific to having the complete 4PT12 as it's impactful enough to warrant being distinct.

Reply:
@TheBackstabi Ok, was just making sure you didn't do it because you thought it might not be possible in the current setup :D